src/transaction/CreateTransaction.tsx

```diff
@@ -79,19 +103,45 @@ async function TestTransaction(
         contractOrEOA,
       };
     }
-  }
 
-  // TODO: Have some sort of immediate feeback given to the user that informs
-  // them that something wrong was given. onChange, onFocus, onSubmit?
-  if (!isAddressValid) {
-    addressElem.style.border = '5px solid #ff0000';
-  }
+    const toastMsg = document.getElementById('toast-message');
+    toastMsg!.innerHTML = 'You don\'t have an existing wallet to test a transaction.';
+
+    const toast = document.getElementById('errToast');
+    toast!.className = 'toast show';
+    setTimeout(() => {
+      toast!.className = 'toast hide';
+    }, 3000);
 
-  if (!isAmountValid) {
-    amountElem.style.border = '5px solid #ff0000';
+    return null;
+  } catch (e: any) {
+    const toastMsg = document.getElementById('toast-message');
+    const i = String(e).indexOf('(');
+    if (String(e).includes('bad response')) {
+      toastMsg!.innerHTML = 'The simulation ran into a network error. Please try testing the transaction again.';
+    } else if (String(e).includes(ERROR) && i !== -1) {
+      let errMsg = String(e).substring(ERROR.length, i - 1);
+      errMsg = errMsg[0].toUpperCase().concat(errMsg.substring(1));
+      toastMsg!.innerHTML = errMsg;
+    } else if (String(e).includes(ERROR) && i === -1) {
+      const errMsg = String(e).substring(ERROR.length);
+      toastMsg!.innerHTML = errMsg;
+    } else {
+      toastMsg!.innerHTML = e;
+    }
```

Comment:
Read the instance properties of the Error object (really only `name` and `message` should be used): https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error Following that, all instances of `String(e)` should generally be replaced with `e.message` as I see it. That way, you don't have to do string parsing or any `indexOf` checks.